farao-dichotomy-api/src/main/java/com/farao_community/farao/dichotomy/api/DichotomyEngine.java

```diff
@@ -33,19 +33,27 @@
  * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  */
 public class DichotomyEngine<T> {
-    private static final Logger LOGGER = LoggerFactory.getLogger(DichotomyEngine.class);
     private static final int DEFAULT_MAX_ITERATION_NUMBER = 100;
     private final Index<T> index;
     private final IndexStrategy indexStrategy;
     private final NetworkShifter networkShifter;
     private final NetworkValidator<T> networkValidator;
     private final int maxIteration;
+    private final Logger logger;
 
     public DichotomyEngine(Index<T> index, IndexStrategy indexStrategy, NetworkShifter networkShifter, NetworkValidator<T> networkValidator) {
-        this(index, indexStrategy, networkShifter, networkValidator, DEFAULT_MAX_ITERATION_NUMBER);
+        this(index, indexStrategy, networkShifter, networkValidator, DEFAULT_MAX_ITERATION_NUMBER, LoggerFactory.getLogger(DichotomyEngine.class));
```

Comment:
I m not convinced by the hack. Mixing technical params (logger ) with business ones . Here you are defining the logger as a key param for the process behind,.. which is not.
I would create in a subLevel a pubic class dedicated for business logging, then use it to separate business from techical logs.

Reply:
changed